Nested structure problem fix

MartinLogan opened the pull request:
Based on the failures we were having and the file unexpectedly trying to process random nested structures that happened to be named "resources" I have attempted to address this.

I used the information at [JSON Output Format](https://www.terraform.io/docs/internals/json-format.html#configuration-representation) to determine that the "resources" should be directly below the "root_module" in the configuration. Please let me know if this is an incorrect assumption.

eerkunt commented:
Thanks for the pr @MartinLogan ! 🎉 

I have a feeling removing `seek_key_in_dict()` ( which is a recursive function ) from there will cause more problems. Let's see if the integration tests will pass.

There was a general problem between GitHub <-> Travis CI within our repository. Just fixed that, so status updates should be correct now.

eerkunt commented:
It sounds like this is failing for scenario described in #227

MartinLogan commented:
Hmm - I was a little concerned about that, figuring that the nested search was there for a reason - Unfortunately just searching for "resources" is too generic - Looks like I may need to dive into the nested module calls to find potential parents of resources and then pull resources from each to avoid the deeply nested pieces - Let me look into #227 and see if I can get something that will deal with bith

MartinLogan commented:
I added back the ability look at the nested module -> resources and fixed my tox.ini so I could run the integration tests locally.

Also added a unit test for it.

I suspect my python is a little verbose, so any suggestions there are welcome.

Also, I may have just moved the issue to nested resources with a module identifier with a property of resources - We could add a check to validate that it is a resource record if that comes up

eerkunt commented on terraform_compliance/extensions/terraform.py:
```diff
@@ -142,12 +142,20 @@ def _parse_configurations(self):
-                    self.data[resource['address']] = resource
-                else:
-                    self.configuration['resources'][resource['address']] = resource
+        resources = []
```
You don't need to declare this since you define this 3 lines below.

eerkunt commented:
#316
